src/axom/inlet/docs/sphinx/simple_types.rst

```diff
@@ -0,0 +1,144 @@
+############
+Simple Types
+############
+
+To help structure your input file, Inlet categorizes the information into
+two types: Fields and Tables.
+
+Fields refer to the individual scalar values that are either at the global level or
+that are contained inside of a Table.
+
+Tables can contain multiple Fields, other sub-Tables, as well as a single array or a single dictionary.
+
+.. note::  There is an inherent global Table that holds all global level Fields.  This can be
+  accessed via your `Inlet` class instance.
+
+
+*******
+Fields
+*******
+
+In Inlet, Fields represent an individual scalar value of primitive type.
+There are four supported field types: ``bool``, ``int``, ``double``, and ``string``.
+
+In this example we will be using the following part of an input file:
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_input_start
+   :end-before: _inlet_simple_types_fields_input_end
+   :language: lua
+
+
+Defining And Storing
+--------------------
+
+This example shows how to add the four simple field types with descriptions to the
+input file schema and add their values, if present in the input file, to the Sidre
+DataStore to be accessed later.
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_add_start
+   :end-before: _inlet_simple_types_fields_add_end
+   :language: C++
+
+You can also add default values to Fields to fall back to if they are not defined
+in your input file. The last added Field was intentionally not present in the input file.  Not all
+fields need to be present, unless they are marked required, like ``a_simple_double``.
+
+Accessing
+---------
+
+Accessing field values stored in Inlet can be accessed via their name with the ``[]`` operator.
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_access_start
+   :end-before: _inlet_simple_types_fields_access_end
+   :language: C++
+
+Here is an example of how to check if the input file contained an optional field:
```

Comment:
The distinction b/w the subscript operator and the `contains` function should be more explicit.

What happens if you use the subscript operator on a key that is not present?
Is it an error? Will the code crash? ...

Also, is there still a `get<T>` accessor ?

Reply:
I reworked this example to be more explicit/less confusing hopefully.

As for the `get<T>` operator.  There currently isn't one for the primitive types.  Only the user-defined ones that also have a `FromInlet<T>` instantiation.  This would be nice due to allowing a single line accessing in some cases since the compiler complains about ambiguity on the following line:

```
std::cout << "a_simple_int = " << inlet["a_simple_int"] << std::endl;
```

@joshessman-llnl  thoughts?